src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -87,6 +87,12 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
             return true;
         }
 
+        /// <summary>
+        /// Authenticate the user identity with the identity provider. 
+        ///
+        /// This could be done through a back channel communication with the identity provider. Exception thrown during
+        /// the authenticating should be saved to the AuthenticateResult.
```

Comment:
Thanks for the suggestion. I'll rethink the overall error handling mechanism here.